### PR TITLE
Expose GirderFS via HTTP server

### PR DESCRIPTION
**Pull request by Xarthisius**
The gist of this PR is the addition of HTTP server, which as a service that can a) mount multiple fses based on JSON payload and b) unmount them.  Other notable points:
1. New dockerfile that runs HTTP server
2. Schema for payload is defined in `girderfs/schema.py`
3. Server itself mostly ports all the code related to volumes from gwvolman

### CODE REVIEW

**Review by craig-willis — COMMENTED** [review]
Preliminary testing looks good.  I need to test exhaustively for the different mount types.

I noted one issue with unmounting when there is no data.  Having a GET handler that dumps the state could be handy for troubleshooting or storing state on disk.  

As discussed, maybe there is a way to store the state and make girderfs container restartable.

File: girderfs/server.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                )
+            elif mount["protocol"] == MountProtocols.bind:
+                source = os.path.join(
+                    os.environ["WT_VOLUMES_PATH"], self.source_path_bind(mount["type"])
+                )
+                cmd = f"sudo mount --bind {source} {destination}"
+            elif mount["protocol"] == MountProtocols.passthrough:
+                cmd = (
+                    "passthrough-fuse -o allow_other "
+                    f"--girder-url={self.state['girderApiUrl']}/folder/{gcObj['_id']}/listing "
```
[the remark this answers is not in the document]
Comment (Xarthisius) [reply girderfs/server.py:111]: https://github.com/whole-tale/girderfs/pull/38/commits/5f9e793aa3ba1f93915366d82595d53d68e60f7a

File: girderfs/server.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                    f"--token={self.gc.token} {destination}"
+                )
+
+            subprocess.check_output(cmd, shell=True)
+
+    def delete(self):
+        for mount in self.state["mounts"]:
```
[the remark this answers is not in the document]
Comment (Xarthisius) [reply girderfs/server.py:118]: https://github.com/whole-tale/girderfs/pull/38/commits/5f9e793aa3ba1f93915366d82595d53d68e60f7a

File: girderfs/server.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                    "opts": "-o uid=1000,gid=100,file_mode=0600,dir_mode=2700",
+                    "source": girder_url + self.webdav_url(mount["type"]),
+                }
+                cmd = (
+                    'echo "{user}\n{pass}" | mount.davfs {opts} {source} {destination}'
+                )
+                cmd = cmd.format(**args)
+            elif mount["protocol"] == MountProtocols.girderfs:
+                gcObj, fs_type = self.mounttype_to_fs(mount["type"])
+                if not gcObj:
```
[the remark this answers is not in the document]
Comment (Xarthisius) [reply girderfs/server.py]: https://github.com/whole-tale/girderfs/pull/38/commits/5f9e793aa3ba1f93915366d82595d53d68e60f7a

**Review by craig-willis — APPROVED** [review]

**Pull request merged**